docs/docs/custom-fields.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Custom fields
+tags:
+  - content-type-builder
+  - plugins
+---
+
+# Custom fields
+
+## Summary
+
+Custom fields provide a way to replace the inputs of existing Strapi types to improve the content editing experience.
+
+## Detailed design
+
+A custom field needs to be registered in both the admin and server.
+
+### Server
+
+We need to ensure that an attribute using a custom field is valid, we expose a new `customFields` object with a `register` method on the `Strapi` instance.
+
+Here's how you register a Custom Field on the server:
+
+```ts
+interface CustomFieldServerOptions {
+  // The name of the custom field
+  name: string;
+  // The name of the plugin creating the custom field
+  plugin?: string;
+  // The existing Strapi data type the custom field will use
+  type: string;
+}
+
+strapi.customFields.register(
+  options: CustomFieldServerOptions | CustomFieldServerOptions[]
+);
+```
+
+The custom field will be added to Strapi during the server [register lifecycle](https://docs.strapi.io/developer-docs/latest/developer-resources/plugin-api-reference/server.html#register).
+
+### Admin
+
+Here's how you register a Custom Field on the admin:
+
+```tsx
+
+// You can also pass an array of objects to register several custom fields at once
+app.customFields.register(
+  options: CustomFieldAdminOptions | CustomFieldAdminOptions[]
+);
+
+interface CustomFieldAdminOptions {
+  // The name of the custom field
+  name: string;
+  // The name of the plugin creating the custom field
+  pluginId?: string;
+  // The existing Strapi data type the custom field will use
+  type: string;
+  // The translation for the name
+  intlLabel: IntlObject;
+  // The translation for the description
+  intlDescription: IntlObject;
+  // The icon for the custom field
+  icon?: React.ComponentType;
+  // The components needed to display the custom field in the Content Manager
+  components: {
+    // Input component for the Edit view
+    Input: () => Promise<{ default: React.ReactComponent }>;
+    // Read only component for the List view
+    View: () => Promise<{ default: React.ReactComponent }>;
+  };
+  // The settings to extend in the Content-Type Builder
+  options?: {
+    base: CTBFormSection[];
+    advanced: CTBFormSection[];
+    validator: (args) => object;
+  }
+}
+
+interface IntlObject {
+  id: string;
+  defaultMessage: string;
+}
+
+interface CTBFormSection {
+  sectionTitle: IntlObject;
+  items: CTBFormInput[];
+}
+
+interface CTBFormInput {
+  name: string;
+  description: InltObject;
+  type: string;
+  intlLabel: IntlObject;
+}
+```
+
+For the admin, we will expose a new `customFields` object with a `register` method on the `StrapiApp` instance. The custom field can then be added to Strapi during the admin [bootstrap lifecycle](https://docs.strapi.io/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.html#bootstrap) by providing the following object to `customFields.register()`.
```

Comment:
Same comment: https://docs.strapi.io/developer-docs/latest/development/custom-fields.html#registering-a-custom-field-in-the-admin-panel